query-engine/query-engine/src/server/mod.rs

```diff
@@ -460,3 +385,67 @@ fn err_to_http_resp(err: query_core::CoreError) -> Response<Body> {
 
     Response::builder().status(status).body(body).unwrap()
 }
+
+struct LogCapture {
+    id: String,
+    capture: bool,
+}
+
+impl LogCapture {
+    fn new(id: String, capture: bool) -> Self {
+        Self { id, capture }
+    }
+
+    fn new_from_req(id: String, req: &Request<Body>) -> Self {
+        let should_capture = Self::get_capture_from_header(req);
+        Self::new(id, should_capture)
+    }
+
+    fn id(&self) -> String {
+        self.id.to_string()
+    }
+
+    fn should_capture(&self) -> bool {
+        self.capture
+    }
+
+    fn get_capture_from_header(req: &Request<Body>) -> bool {
+        match req.headers().get("CAPTURE_LOGS") {
```

Comment:
Btw, sorry for bringing this up late, this should've been raised on the tech spec stage, but can we change it to use a dash instead of an underscore? i.e., `capture-logs` or `X-capture-logs` or whatever. Underscores in header names may be problematic as they are considered invalid by lots of software for historical reasons (even though technically legal according to the HTTP standard), and may be dropped along the way by firewalls, reverse proxies etc. Notably, nginx drops them unless `underscores_in_headers on;` is set in configuration.

Reply:
Good catch. Should we make this `PRISMA-CAPTURE-LOGS`?

Reply:
`Prisma-Capture-Logs` sounds great, I think it's the best option. But we should probably use a consistent naming scheme for all headers.

So far we have:

* `X-transaction-id`
* `PRISMA-CAPTURE-LOGS`
* `traceparent`

Reply:
(Not super helpful comment re `X-` usage: https://tonyxu.io/posts/2018/http-deprecate-x-prefix/ TLDR: Not recommended any more - but many of course still do it)

Reply:
> (Not super helpful comment re X- usage: https://tonyxu.io/posts/2018/http-deprecate-x-prefix/ TLDR: Not recommended any more - but many of course still do it)

My impression was it's more about extensions to protocols than application-specific headers, because the motivation for that RFC was that non-standard headers that many find useful tend to eventually become standard headers, which doesn't really apply in this case, but yeah, I wouldn't even mention `X-` if we didn't already have one header with that prefix.

Reply:
That's why this is not suuuuper helpful :D

I think we can do whatever here really, using the `x-` would just continue what we already have.